src/Syntax/Implicits.hs

```diff
@@ -45,7 +45,7 @@ deriving instance Eq (Var p) => Eq (Obligation p)
 
 -- | A concrete representation of /one/ implicit parameter, stored in an
 -- implicit parameter scope (a trie, indexed by the 'head').
-data Implicit p
+data Implicit info p
```

Comment:
This always seems to be `ImplicitScope` - can we remove it at all, or is it just to break a dependency cycle?

Reply:
It's open just to break the cycle, yeah